packages/polymath-ui/src/components/EthNetworkWrapper/__tests__/EthNetworkWrapper.js

```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+
+import { EthNetworkWrapper } from '../EthNetworkWrapper';
```

Comment:
Export components and test them "not Redux connected" is a solution but I wouldn't say it's "the best solution" to the point of having to refactor this (see [comment of Enzyme author saying you shouldn't do it](https://github.com/airbnb/enzyme/issues/1002#issuecomment-340065900)).
I usually mock the Redux store <Provider> because I can also provide initial state in my tests to get snapshots of different scenarios (component with different initial props) and all of this going through Redux mapStateToProps just like on the real app.
It also allows you to snapshot an entire page without having bugs if a descendant component is connected (for certain components that you don't want to shallow render).
Overall I think it's fine for polymath-ui to do this but for issuer I would definitely mock the Redux store.

Reply:
If you need to test multiple components rendered in the same test then that's fine, mocking the store sounds good.

For `Container` tests I'm not convinced by the argument (it has its good amount of downvotes even). The argument "you'd never use it on its own on production" doesn't make much sense if we talk about **unit**  tests. In unit tests we want isolation.

That being said, I think testing the component wrapped on a `Provider` component is fine if the store is mocked. So if it makes a test simpler I have no issue with that, its a matter of trade-offs. One requires mocking the HOCs the other requires exporting the different pieces of *our* code only and mock the APIs.

Bottomline:
**Let's do whatever is simple as long as we are able to test properly and tests are not too fragile**

P.S: I removed the global mocked store until we actually need something like that, I do think that could be dangerous if it is not reset after each test, and a utility function to me feels safer than a global object shared across tests (for instance, what happens if tests run in parallel and mutate this store?).